app/controllers/api/v1/search_controller.rb

```diff
@@ -557,24 +565,48 @@ def self.generate_mongo_query_by_context(terms:, base_studies:, accessions:, que
           results_matched_by_data['numResults:scp'] = studies.length # Total number of SCP results
           # Azul study results to be added with SCP-4202
 
-          return {:studies => studies, :results_matched_by_data => results_matched_by_data}
+          { studies: studies, results_matched_by_data: results_matched_by_data, metadata_matches: metadata_matches }
 
         when :inferred
           # in order to maintain the same behavior as normal facets, we run each facet separately and get matching accessions
           # this gives us an array of arrays of matching accessions; now find the intersection (:&)
-          filters = terms.values.map {|keywords| escape_terms_for_regex(term_list: keywords)}
-          accessions_by_filter = filters.map {|filter| base_studies.any_of({name: filter}, {description: filter})
-                                                           .where(:accession.nin => accessions).pluck(:accession)}
+          filters = terms.values.map { |keywords| escape_terms_for_regex(term_list: keywords) }
+          accessions_by_filter = filters.map {|filter| base_studies.any_of({ name: filter }, { description: filter })
+                                                                   .where(:accession.nin => accessions)
+                                                                   .pluck(:accession) }
 
           studies = base_studies.where(:accession.in => accessions_by_filter.inject(:&))
-          return {:studies => studies, :results_matched_by_data => {}}
+          return { studies: studies, results_matched_by_data: {} }
 
         else
           # no matching query case, so perform normal text-index search
-          studies= base_studies.any_of({:$text => {:$search => terms}}, {:accession.in => accessions})
-          return {:studies => studies, :results_matched_by_data => {}}
-    
+          studies = base_studies.any_of({ :$text => { :$search => terms } }, { :accession.in => accessions })
+          { studies: studies, results_matched_by_data: {}, metadata_matches: {} }
+        end
+      end
+
+      # take the output from :match_facet_filters_from_keywords and match to any possible CellMetadatum in the database
+      # will return a hash of accessions to matched metadata entries
+      def self.get_studies_from_term_conversion(terms)
+        accessions_to_filters = {}
```

Comment:
This method also should go in `app/lib/study_search_service.rb` and we should have independent test cases for it